tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,639 @@
+
+    public TspClientUpdateTool(ILogger<TspClientUpdateTool> logger, IOutputService output)
+    {
+        this.logger = logger;
+        this.output = output;
+    }
+
+    public override Command GetCommand()
+    {
+        var root = new Command("customized-update", "Workflow for updating customized TypeSpec-generated client code");
+
+        var regenerate = new Command("regenerate", "Generate new client code artifacts for comparison");
+        regenerate.AddArgument(_specPathArg);
+        regenerate.AddOption(_generatedNewOpt);
+        regenerate.AddOption(_sessionIdOpt);
+        regenerate.SetHandler(async ctx => { await HandleRegenerate(ctx, ctx.GetCancellationToken()); });
+
+        var diff = new Command("diff", "Detect API surface changes between old and new generation outputs");
+        diff.AddOption(_sessionIdOpt);
+        diff.AddOption(_generatedOldOpt);
+        diff.AddOption(_generatedNewOpt);
+        diff.SetHandler(async ctx => { await HandleDiff(ctx, ctx.GetCancellationToken()); });
+
+        var map = new Command("map", "Map API changes to impacted customization files");
+        map.AddOption(_sessionIdOpt);
+        map.SetHandler(async ctx => { await HandleMap(ctx, ctx.GetCancellationToken()); });
+
+        var merge = new Command("merge", "Apply direct merges for unaffected customization files");
+        merge.AddOption(_sessionIdOpt);
+        merge.SetHandler(async ctx => { await HandleMerge(ctx, ctx.GetCancellationToken()); });
+
+        var propose = new Command("propose", "Propose patch diffs for impacted customization files");
+        propose.AddOption(_sessionIdOpt);
+        propose.AddOption(_filesOpt);
+        propose.SetHandler(async ctx => { await HandlePropose(ctx, ctx.GetCancellationToken()); });
+
+        var apply = new Command("apply", "Apply previously proposed patches to customization files");
+        apply.AddOption(_sessionIdOpt);
+        apply.AddOption(_dryRunOpt);
+        apply.SetHandler(async ctx => { await HandleApply(ctx, ctx.GetCancellationToken()); });
+
+        var status = new Command("status", "Show current session status and counts");
+        status.AddOption(_sessionIdOpt);
+        status.SetHandler(async ctx => { await HandleStatus(ctx, ctx.GetCancellationToken()); });
+
+        root.AddCommand(regenerate);
+        root.AddCommand(diff);
+        root.AddCommand(map);
+        root.AddCommand(merge);
+        root.AddCommand(propose);
+        root.AddCommand(apply);
+        root.AddCommand(status);
+
+        CommandHierarchy = [ SharedCommandGroups.EngSys ];
+        return root;
+    }
+
+    public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct) => await Task.CompletedTask;
+
+    // --------------- MCP Methods ---------------
+    [McpServerTool(Name = "tsp_update_regenerate"), Description("Generate new code artifacts and start (or continue) an update session")] 
```

Comment:
Still in flight but we're going to prefix all tool names with `azsdk_` (https://github.com/Azure/azure-sdk-tools/pull/11619/files).